src/telem/lib/output/GraphiteOutputAdapter.lua

```diff
@@ -0,0 +1,63 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local OutputAdapter     = require 'telem.lib.OutputAdapter'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local GraphiteOutputAdapter = o.class(OutputAdapter)
+GraphiteOutputAdapter.type = 'GraphiteOutputAdapter'
+
+-- Graphite Output Adapter
+--
+-- Endpoint: Grafana Cloud or hosted Graphite metric endpoint
+-- API Key: Token with at least metric:write scope
+--
+-- Note that API key are prefixed with a numeric, i.e. 2021620:your-api-key
+function GraphiteOutputAdapter:constructor(endpoint, apiKey)
+    self:super('constructor')
+    self.endpoint = assert(endpoint, 'Endpoint is required')
+    self.apiKey   = assert(apiKey,   'API key is required')
+end
+
+function GraphiteOutputAdapter:write(collection)
+    assert(o.instanceof(collection, MetricCollection),
+           'Collection must be a MetricCollection')
+
+    -- Build an array of metric objects
+    local payload = {}
+    -- Current unix timestamp
+    local now = math.floor(os.epoch('utc') / 1000)
+
+    for _, metric in pairs(collection.metrics) do
+        local tags = {}
+        if metric.unit    and metric.unit    ~= '' then table.insert(tags, 'unit='   .. metric.unit)    end
+        if metric.source  and metric.source  ~= '' then table.insert(tags, 'source=' .. metric.source)  end
+        if metric.adapter and metric.adapter ~= '' then table.insert(tags, 'adapter='.. metric.adapter) end
+        
+        -- Graphite metric name are namespaced by dots "."
+        local normalized_name = string.gsub(metric.name, ':', '.')
+
+        table.insert(payload, {
+            name     = normalized_name,
+            interval = 30,
+            value    = metric.value,
+            tags     = tags,
+            time     = metric.time or now,

```

Comment:
Is `metric.time` this something that you are using on your end? This isn't a standard Metric attribute, though it is something I have considered as part of a larger normalization. I don't mind it being here, just curious.